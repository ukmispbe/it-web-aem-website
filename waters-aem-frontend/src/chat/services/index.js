import 'whatwg-fetch';

class ChatService {
    constructor(
        countryCode = 'US',
        statusApi = 'https://stgservices.waters.com/api/waters/v1/chat/enabled/{countryCode}'
    ) {
        this.countryCode = countryCode;
        this.statusApi = statusApi;
    }

    checkFetch(response) {
        if (!response.ok){
            throw response;
        }
        return response;
    }

    getChatStatus() {
        const url = this.statusApi.replace('{countryCode}', this.countryCode);
        return this.getData(url);
    }

    getData(url){
        return new Promise((resolve, reject) => {
            window
                .fetch(url)
                .then(this.checkFetch)
                .then(response => {
                    resolve(response.json())
                })
                .catch(err => {
                    reject(err);
                });
        });
    }
}

export default ChatService;