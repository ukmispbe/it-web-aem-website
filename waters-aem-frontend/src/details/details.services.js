import { fetch } from 'whatwg-fetch';
import { getCategoryReferenceType } from '../utils/userFunctions';

const getData = async (url) => {
    const response = await fetch(url, {
        method: 'GET',
        credentials: 'include',
        headers: {
            'Content-Type': 'application/json'
        }
    });
    return await response;
}

export const getOrderDetails = async (endpoint, id, setError) => {
    const url = endpoint + "/" + id;

    const response = await getData(url);
    const responseBody = await response.json();

    if(response.status === 200) {
        return responseBody;
    } else {
        setError({status: response.status, code: responseBody.code});
    }
}

export const getQuoteDetails = async (endpoint, setError) => {
    const response = await getData(endpoint);
    const responseBody = await response.json();

    if(response.status === 200) {
        return responseBody;
    } else {
        setError({status: response.status, code: responseBody.code});
    }
}

const buildSearchURL = (endpoint, lineItems, isocode) => {
    let skus, rows, keywords, url;
    skus = lineItems.map(lineItem => lineItem.materialNumber);
    rows = skus.length;
    keywords = skus.join(' ');
    return url = `${endpoint}/category_facet$shop:Shop?keyword=${keywords}&rows=${rows}&isocode=${isocode}&multiselect=true&page=1&sort=most-relevant${getCategoryReferenceType()}`;
}

export const getItemDetails = async (endpoint, lineItems, setError, isocode) => {
    const url = buildSearchURL(endpoint, lineItems, isocode);
    const response = await getData(url);
    const responseBody = await response.json();

    if(response.status === 200) {
        return responseBody;
    } else {
        setError({status: response.status, code: responseBody.code});
    }
}

export const matchLineItems = (orderDetailsAPIResults, searchAPIResults) => {
    orderDetailsAPIResults.lineItems.forEach(soldItem => {
        for (let i = 0; i < searchAPIResults.length; i++) {
            if(soldItem.materialNumber === searchAPIResults[i].skucode) {
                soldItem.url = searchAPIResults[i].url;
                soldItem.title = searchAPIResults[i].title;
                soldItem.description = searchAPIResults[i].description;
                soldItem.thumbnail = searchAPIResults[i].thumbnail;
            } 
        }
    });
    
    return orderDetailsAPIResults;
}