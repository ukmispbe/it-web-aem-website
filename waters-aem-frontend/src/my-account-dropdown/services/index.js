import 'whatwg-fetch';
import SessionStore from '../../stores/sessionStore';
import LocalStore from '../../stores/localStore';
import {setUserDataForDataDog} from '../../utils/userFunctions'

const getData = async (url) => {
    const response = await fetch(url, {
        method: 'GET',
        credentials: 'include',
        headers: {
            'Content-Type': 'application/json'
        }
    });
    return response;
};

export async function signOutRequest(url, signOutUrl, homepageLink) {
    const sessionStore = new SessionStore();
    sessionStore.removeUserDetails();
    sessionStore.removeSoldToDetails();
    const localStore = new LocalStore();
    localStore.removeGUID();
    localStore.removeCartId();

    const response = await getData(url);
    //TODO: Remove after MyAccount-R-6.1.0 release

    if(signOutUrl.indexOf('/nextgen') === -1) {
        homepageLink = signOutUrl;
    }
    setUserDataForDataDog();
    window.location.href = homepageLink;
}