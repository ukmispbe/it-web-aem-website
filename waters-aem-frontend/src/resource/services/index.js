/**
 * Builds Resource data
 * @param {Array} data 
 * @returns Array
 */
const formData = data => {
    const { num_found } = data;
    if (num_found > 0) {
        const { documents } = data;
        return documents.reduce((acc, item) => {
            acc.push({
                title: item.title,
                url: item.url,
                yearpublished_facet: item.yearpublished_facet ? item.yearpublished_facet.replace('[', '').replace(']', '') : ''
            });
            return acc;
        }, []);
    } else {
        return [];
    }
}

/**
 * Fetches data from given url
 * @param {String} url 
 * @returns {Object} 
 */
const getResourcesData = async url => {
    const callService = window.fetch(url).then(res => res.json());
    const data = await callService;
    const response = { numFound: data.num_found };
    response.documents = formData(data);
    return response;
};

export default getResourcesData;