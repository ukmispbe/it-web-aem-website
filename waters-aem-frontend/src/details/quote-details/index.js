import React, { Component } from 'react';
import ReactSVG from 'react-svg';
import { getQuoteDetails } from '../details.services';
import { buildViewCartURL }from '../../utils/eCommerceFunctions';
import Shipment from '../components/shipment';
import Spinner from "../../utils/spinner";
import ErrorBoundary from '../../search/ErrorBoundary';
import Modal, { Header, keys } from '../../utils/modal';
import AddToCartBody from '../../sku-details/views/addToCartModal';
import Analytics, { analyticTypes } from '../../analytics';
import { DELIVERY_STATUS, STORE, CHECKOUT } from '../../constants';
import DeliveryStatus from '../../common/delivery-status';
import { getFullCompanyAddress, getCartCheckoutUrl, getUrlPath, getUrlParameter, convertToBoolean, getApprovalStatus } from '../../utils/userFunctions';
import SessionStore from '../../stores/sessionStore';

class QuoteDetails extends Component {
    constructor({setErrorBoundaryToTrue, resetErrorBoundaryToFalse, removeNotifications, ...props}) {
        super({setErrorBoundaryToTrue, resetErrorBoundaryToFalse, removeNotifications, ...props});
        this.state = {
            quoteId: getUrlParameter("id"),
            detailsUrl: props.config.fetchDetailsEndPoint,
            itemsUrl: props.config.fetchItemsEndPoint,
            quoteDetails: {},
            errorServiceError: false,
            errorOrderNotFound: false,
            isLoading: true,
            modalShown: false,
            modalConfig: props.config.modalInfo,
            totalItemsCount:0
        }
        this.page = {
            name: "Quote Details",
            type: "Quotes",
            analytics: {
                reference: "quoteDetails",
            }
        }
    }

    rootStyle = "cmp-order-details";
    setAnalytics = (event, detail={}) => {
        const model = {
            detail,
            event
        };
        Analytics.setAnalytics(analyticTypes[this.page.analytics.reference].name, model);
    }

    toggleModal = () => {
        this.setState({ modalShown: !this.state.modalShown });
    };

    setError = (response) => {
        this.setAnalytics('error', {error: {...response}});
        if (response.status === 400 && response.code === 704) {
            this.setState({orderNotFoundError: true});
        } else {
            this.props.setErrorBoundaryToTrue({code: 400});
            this.setState({errorServiceError: true});
        }
    }

    getQuoteDetailsData = () => {
        const { detailsUrl, quoteId } = this.state;       
        const url = getQuoteDetailsUrl(detailsUrl, quoteId);
        getQuoteDetails(url, this.setError)
            .then((data) => {
                const quotes = data && data.quotes || undefined;
                let totalItemsCount = 0;
                if(quotes) {
                    totalItemsCount = quotes.totalItems || 0;
                    this.setState({
                        isLoading: false,
                        quoteDetails: quotes,
                        totalItemsCount,
                        errorOrderNotFound:false
                    });
                } else {
                    this.setState({
                        errorOrderNotFound: true,
                        isLoading: false,
                        totalItemsCount
                    });
                }
            })
    }

    async componentDidMount() {
        this.getQuoteDetailsData();

        const commerceConfig = JSON.parse(
            document.getElementById('commerce-configs-json').innerHTML
        );
        if(commerceConfig) {
            this.setState({
                addToCartUrl: commerceConfig.addToCartUrl,
                viewCartUrl: buildViewCartURL(commerceConfig.viewCartUrl)
            });
            // Update modal config button with a callback and new cart url
            const buttons = [...this.state.modalConfig.buttons];
            buttons[0] = {
                ...buttons[0],
                action: buildViewCartURL(commerceConfig.viewCartUrl),
                callback: "" //this.addToCartReorder
            }
            const updatedModalConfig = {
                ...this.state.modalConfig,
                buttons: buttons
            }
            this.setState({
                modalConfig: updatedModalConfig
            })
        }
    }

    static getDerivedStateFromProps(nextProps, prevState){
        const {quoteId} = prevState;
        const urlQuoteId = getUrlParameter("id");
        if(quoteId !== urlQuoteId){
            return {quoteId:urlQuoteId}
        }
    }

    componentDidUpdate(prevProps, prevState){
        const {quoteId} = prevState;
        if(quoteId !== this.state.quoteId){
            this.getQuoteDetailsData();
        }
    }

    componentWillUnmount() {
        this.props.resetErrorBoundaryToFalse();
        this.props.removeNotifications();
    }

    config = document.getElementById('json-config--cmp-detail-tiles--personal')
        ? JSON.parse(document.getElementById('json-config--cmp-detail-tiles--personal').innerHTML) : '';

    renderAddress = (address = {}) => {
        if(address){
            const addressArray = getFullCompanyAddress(address, true);
            return (
                <>
                    {addressArray.map((addressLine) => <div className={`${this.rootStyle}-address1`} data-locator="order-details-address">{addressLine}</div>)}
                </>
            );
        }
        
    }

    renderItemCount = () => {
        const {totalItemsCount } = this.state;
        let label = "";
        if (totalItemsCount > 0) {
            if (totalItemsCount > 1) {
                label = this.props.config.items;
            } else if (totalItemsCount === 1) {
                label = this.props.config.item;
            }

            let itemCountLabel =  " (" + totalItemsCount + " " + label + ")";
            return itemCountLabel;

        } else {
            return label;
        }
    }

    placeOrderForQuote = (e, quoteId) => {
	   e.preventDefault();
       if(quoteId){		
        (new SessionStore()).setQuoteId(quoteId);    
        const {quoteDetails} = this.state;
        const {totalItems, entries, subTotal, totalShippingAndHandling, totalDiscounts, totalTax, totalPriceWithTax = {}} = quoteDetails
        const placeOrderModel ={
            detail:{
                quoteId,
                subTotal,
                totalTax,
                totalDiscounts,
                totalShippingAndHandling,
                totalPriceWithTax,
                totalItems,
                entries,
            }
        }
        this.setAnalytics('quotePlaceOrder', placeOrderModel)      
		const checkoutUrl =  getCartCheckoutUrl(STORE,CHECKOUT);
        setTimeout(()=>{ window.location.href = checkoutUrl; }, 500);
        
       }
    }

    renderPlaceOrderButton = (className,elementLocator) => {
        const {quoteDetails} = this.state;
        const {quoteStatus, quoteId} = quoteDetails;
        const {config} = this.props;
        const {reorderTitle,isQuoteToOrderDisabled} = config;
        let commerceConfigs = document.getElementById('commerce-configs-json');
        if(commerceConfigs){
            commerceConfigs = JSON.parse(commerceConfigs.innerHTML);
        }
        const {isQuoteDisabled,isCheckoutDisabled} = commerceConfigs;
        const approvalStatus = getApprovalStatus()
        const isDisabled = convertToBoolean(isQuoteDisabled) || convertToBoolean(isCheckoutDisabled) || approvalStatus === 'R' || isQuoteToOrderDisabled;
        return quoteStatus === DELIVERY_STATUS.OPEN && !isDisabled && (
            <div className={className} data-locator={elementLocator}>
                <a className="cmp-button" href="#" onClick={(e) => this.placeOrderForQuote(e,quoteId)} data-locator={`${elementLocator}-button`} >
                    {reorderTitle}
                </a>
            </div>
        )
    }

    quoteAgain = (e, quoteId) => {
        e.preventDefault();
        if(quoteId){
         const quoteAgainModel ={
             detail:{
                 quoteId,
             }
         }
         this.setAnalytics('quoteAgainClick', quoteAgainModel)       
        }
    }

    renderQuoteAgainButton = className => {
        const {quoteDetails} = this.state;
        const {quoteStatus, quoteId} = quoteDetails;
        return quoteStatus === DELIVERY_STATUS.EXPIRED && (
            <div className={className} data-locator="quote-details-quote-again-cta">
                <a className="cmp-button" href="/#" onClick={(e) => this.quoteAgain(e,quoteId)} >
                    {this.props.config.quoteAgainTitle}
                </a>
            </div>
        )
    }

    getValue = (data,attribute,defaultValue = undefined) => {
        let value = defaultValue
        if(data){
           value = data[attribute];
        }
       return value
    }
    
    renderDetailsSection = () => {
        const { quoteDetails } = this.state;
        const { config } = this.props;
        const {created, expires, shipTo, billTo,savings,shipping, tax, totalLabel,shipment,icons, isShowQuoteAgainButton} = config;
        const {quoteId,quoteCreationDate,quoteExpirationDate, subTotal,totalShippingAndHandling,totalDiscounts,totalTax,totalPriceWithTax, shipToInfo, billToInfo,quoteStatus, orderNumber, replacedQuoteNumber} =  quoteDetails;
        const isTotalDiscount = this.getValue(totalDiscounts,'value', '0');
        const notZeroDiscountFlag = parseFloat(isTotalDiscount) !== 0 ? true : false;
        const subTotalValue = this.getValue(subTotal,'formattedValue');
        const ShippingAndHandlingValue = this.getValue(totalShippingAndHandling,'formattedValue');
        const totalDiscountsValue = this.getValue(totalDiscounts,'formattedValue');
        const totalTaxValue = this.getValue(totalTax,'formattedValue');
        const totalPriceValue = this.getValue(totalPriceWithTax,'formattedValue');
        const showExpireDate = !!(quoteStatus === DELIVERY_STATUS.PENDING || quoteStatus === DELIVERY_STATUS.REJECTED || quoteStatus === DELIVERY_STATUS.OPEN);
        const showNewDetailsLinkSection = (quoteStatus === DELIVERY_STATUS.QUOTE_REPLACED || quoteStatus === DELIVERY_STATUS.ORDER_PLACED);
        const newItemUrl = quoteStatus === DELIVERY_STATUS.ORDER_PLACED ? `#orderdetails?id=${orderNumber}` : `#quotedetails?id=${replacedQuoteNumber}`;
        return (<>
            <div className={`${this.rootStyle}__container`}>
                <h2 className={`${this.rootStyle}__title`} data-locator="product-title">
                {config.detailsTitle}
                </h2>
                {showNewDetailsLinkSection && (<div className={`${this.rootStyle}__new-details-link-text`}>
                    <div className="new-details-link-section">
                    <a href={newItemUrl}>
                        <div className="new-details-icon">
                            <ReactSVG src={config.icons.newQuoteOrderIcon} />
                        </div>
                        {quoteStatus === DELIVERY_STATUS.QUOTE_REPLACED && (<div className="new-details-text" data-locator="delivery-text">
                            {`${config.newQuote}${replacedQuoteNumber}`}
                        </div>)}
                        {quoteStatus === DELIVERY_STATUS.ORDER_PLACED && (<div className="new-details-text" data-locator="delivery-text">
                            {`${config.orderNumberText}${orderNumber}`}
                        </div>)}
                    </a>    
                    </div>
                    {quoteStatus && quoteStatus === DELIVERY_STATUS.ORDER_PLACED && (<div className="new-details-status-icon"><DeliveryStatus
                        status={quoteStatus}
                        labels={shipment}
                        icons={icons}
                    /> </div>)} 
                </div>)}
                <div className={`${this.rootStyle}__order-info`}>
                <h3 className={`${this.rootStyle}__order-number`} data-locator="product-number">
                        {config.numberLabel + ": " + quoteId}
                    </h3>
                </div>
                <div className={`${this.rootStyle}__order-summary`}>
                {quoteStatus && quoteStatus !== DELIVERY_STATUS.ORDER_PLACED && (<DeliveryStatus
                        status={quoteStatus}
                        labels={shipment}
                        icons={icons}
                    />)} 
                </div>
                <div className={`${this.rootStyle}__order-info`}>
                    {quoteCreationDate && (<div className={`${this.rootStyle}__order-date`} data-locator="quote-details-created-date">
                        {`${created} ${quoteCreationDate}`}
                    </div>)}
                    {showExpireDate && quoteExpirationDate && (<div className={`${this.rootStyle}__order-date`} data-locator="quote-details-expire-date">
                    {`${expires} ${quoteExpirationDate}`}
                    </div>)}
                    <div className={`${this.rootStyle}__address-container`}>
                        {shipToInfo && (<div className={`${this.rootStyle}__ship-to`} data-locator="ship-to">
                            <h4>{shipTo}</h4>
                            {this.renderAddress(shipToInfo)}
                        </div>)}
                        {billToInfo && (<div className={`${this.rootStyle}__bill-to`} data-locator="bill-to">
                            <h4>{billTo}</h4>
                            {this.renderAddress(billToInfo)}
                        </div>)}
                    </div>
                </div>
                <div className={`${this.rootStyle}__order-summary`} data-locator="order-summary-cart-details">
                    <h4>{config.summaryTitle}</h4>
                    <div className={`${this.rootStyle}__order-subtotal`} data-locator="order-summary-line-sub-total">
                        <div className={`${this.rootStyle}__order-subtotal_left`} data-locator="order-summary-label-sub-total">{config.subTotal} {this.renderItemCount()}</div>
                        <div className={`${this.rootStyle}__order-subtotal_right`} data-locator="order-summary-price-sub-total">{subTotalValue}</div>
                    </div>
                    {notZeroDiscountFlag && 
                        <div className={`${this.rootStyle}__order-savings`} data-locator="order-summary-line-total-discount">
                            <div className={`${this.rootStyle}__order-savings_left`} data-locator="order-summary-label-total-discount">{savings}</div>
                            <div className={`${this.rootStyle}__order-savings_right`} data-locator="order-summary-price-total-discount">{config.minusSign}{totalDiscountsValue}</div>
                        </div>
                    }
                    <div className={`${this.rootStyle}__order-shipping`} data-locator="order-summary-line-total-shipping-handling">
                        <div className={`${this.rootStyle}__order-shipping_left`} data-locator="order-summary-label-total-shipping-handling">{shipping}</div>
                        <div className={`${this.rootStyle}__order-shipping_right`} data-locator="order-summary-price-total-shipping-handling">{ShippingAndHandlingValue}</div>
                    </div>
                    <div className={`${this.rootStyle}__order-tax`} data-locator="order-summary-line-estimated-tax">
                        <div className={`${this.rootStyle}__order-tax_left`} data-locator="order-summary-label-estimated-tax">{tax}</div>
                        <div className={`${this.rootStyle}__order-tax_right`} data-locator="order-summary-price-estimated-tax">{totalTaxValue}</div>
                    </div>
                    <div className={`${this.rootStyle}__order-total`} data-locator="order-summary-line-total-price">
                        <div className={`${this.rootStyle}__order-total_left`} data-locator="order-summary-label-total-price">{totalLabel}</div>
                        <div className={`${this.rootStyle}__order-total_right`} data-locator="order-summary-price-total-price"><h1>{totalPriceValue}</h1></div>
                    </div>
                    {this.renderPlaceOrderButton(`${this.rootStyle}__reorder`,"quote-details-summary-order-place")}
                    {isShowQuoteAgainButton && this.renderQuoteAgainButton(`${this.rootStyle}__reorder`)}
                </div>
            </div>
            </>
        )
    }

    renderNotFoundError = () => {
        return (
            <>
                <div className="cmp-order-details__no-results" data-locator="order-details-no-results">
                    <p>{this.props.config.resultNotFoundErrorTitle}</p>
                </div>
            </>
        );
    }

    renderOrderShipmentList = () => {
        const { quoteDetails,totalItemsCount } = this.state;
        const {entries = []} = quoteDetails;
        const {isShowQuoteAgainButton} = this.props.config;
        if(entries && entries.length > 0){
            return (
                <>
                    <div className="cmp-order-details__order-shipment-list" data-locator="order-shipment-list">
                        <hr className="order-shipment-list__hr"/>
                        <Shipment
                            data={entries}
                            shipment={this.props.config.shipment}
                            icons={this.props.config.icons}
                            resultsText={this.props.config.resultsText}
                            noResultsFoundTitle={this.props.noResultsFoundTitle}
                            totalItemsOrdered={totalItemsCount}
                            totalItems={entries.length}
                            isQuoteDetails={true}
                        />
                    </div>
                    {this.renderPlaceOrderButton("order-shipment__reorder","quote-details-shipment-order-place")}
                    {isShowQuoteAgainButton && this.renderQuoteAgainButton("order-shipment__reorder")}
                </>
            )
        }
        else{
            return null;
        }
    }

    render() {
        const { isLoading, errorOrderNotFound, errorServiceError } = this.state;
        return (
            <>
                {isLoading && (<Spinner loading={isLoading} />)}
                {!isLoading && errorOrderNotFound && this.renderNotFoundError()}
                {!errorOrderNotFound && !errorServiceError && !isLoading && this.renderDetailsSection()}
                {!errorOrderNotFound && !errorServiceError && !isLoading && this.renderOrderShipmentList()}
                <Modal isOpen={this.state.modalShown} onClose={this.toggleModal} className='cmp-add-to-cart-modal'>
                    <Header
                        title={this.state.modalConfig.title}
                        icon={this.state.modalConfig.icon}
                        className={keys.HeaderWithAddedMarginTop}
                    />
                    <AddToCartBody
                        config={this.state.modalConfig}
                        errorObjCart={this.state.errorObjCart}
                    />
                </Modal>
            </>
        )
    }
}

const ErrorBoundaryQuoteDetails = props => (
    <ErrorBoundary>
        <QuoteDetails {...props} />
    </ErrorBoundary>
)
export { QuoteDetails };
export default ErrorBoundaryQuoteDetails;