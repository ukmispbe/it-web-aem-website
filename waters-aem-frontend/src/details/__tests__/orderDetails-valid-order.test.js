import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { OrderDetails } from '../order-details/index';
import * as getOrderDetails from '../details.services';
import props from '../__mocks__/en_US/index';
import { orderDetailsJSON } from '../__mocks__/en_US/services-json.test';
import mockBodyHTML from '../../__mocks__/en_US/html/mock-body-html'

describe('Feature: Order Details Component', () => {
    let wrapper;

    beforeAll(async () => {
        delete window.location;
        window.location = new URL('https://www.waters.com/nextgen/us/en/account/my-account.html#orderdetails?id=15740002');
        window.scrollTo = jest.fn();
        global.fetch = jest.fn();
    });

    beforeEach(() => {
        const setErrorBoundaryToTrue = jest.fn();
        const resetErrorBoundaryToFalse = jest.fn();
        const removeNotifications = jest.fn();
        wrapper = shallow(<OrderDetails {...props} setErrorBoundaryToTrue={setErrorBoundaryToTrue} resetErrorBoundaryToFalse={resetErrorBoundaryToFalse} removeNotifications={removeNotifications} />, { disableLifecycleMethods: true });
    });

    afterEach(() => {
        wrapper.unmount();
        jest.clearAllMocks();
    });

    describe('Scenario: Rendering', () => {
        document.body.innerHTML = mockBodyHTML;

        describe("When component is mounted", () => {
            it("It should get order id from url", async () => {
                expect(window.location.hash).toEqual("#orderdetails?id=15740002");
            });

            it('should fetch data from server and set orderDetails state', done => { // 1

                const spyDidMount = jest.spyOn(OrderDetails.prototype,"componentDidMount");
                const spyGetOrderDetails = jest.spyOn(getOrderDetails, 'getOrderDetails').mockImplementation(() => {
                    return Promise.resolve(orderDetailsJSON);
                });
                const didMount = wrapper.instance().componentDidMount();
                expect(spyDidMount).toHaveBeenCalled();

                didMount.then(() => {
                    wrapper.update();
                    expect(spyGetOrderDetails).toHaveBeenCalled();
                    expect(wrapper.state('orderId')).toBe('15740002');
                    expect(wrapper.state('orderDetails')).toBe(orderDetailsJSON);

                    spyDidMount.mockRestore();
                    fetch.mockClear();
                    done();
                });
            });

            it('Then the snapshot should match', async () => {
                let json;
                await renderer.act(async () => {
                    json = renderer.create(<OrderDetails {...props} />);
                });
                expect(json).toMatchSnapshot();
            });

        })
    })
})