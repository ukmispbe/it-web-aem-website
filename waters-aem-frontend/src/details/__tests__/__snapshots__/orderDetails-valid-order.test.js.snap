// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Order Details Component Scenario: Rendering When component is mounted Then the snapshot should match 1`] = `
Array [
  <div
    className="cmp-order-details__container"
  >
    <h2
      className="cmp-order-details__title"
      data-locator="product-title"
    >
      Order Details
    </h2>
    <div
      className="cmp-order-details__order-info"
    >
      <h3
        className="cmp-order-details__order-number"
        data-locator="product-number"
      >
        Order Number: 15740002
      </h3>
      <div
        className="cmp-order-details__order-date"
        data-locator="order-date"
      >
        February 18, 2020
      </div>
      <div
        className="cmp-order-details__address-container"
      >
        <div
          className="cmp-order-details__ship-to"
          data-locator="ship-to"
        >
          <h4>
            Ship to
          </h4>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            Neta Scientific Inc
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            4206 Sylon Blvd
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            Hainesport, NJ 08036-3736
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            United States
          </div>
        </div>
        <div
          className="cmp-order-details__bill-to"
          data-locator="bill-to"
        >
          <h4>
            Bill to
          </h4>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            Neta Scientific Inc
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            4206 Sylon Blvd
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            Hainesport, NJ 08036-3736
          </div>
          <div
            className="cmp-order-details-address1"
            data-locator="order-details-address"
          >
            United States
          </div>
        </div>
      </div>
      <div
        className="cmp-order-details__payment-container"
      >
        <div
          className="cmp-order-details__payment-method"
          data-locator="payment-method"
        >
          <h4>
            Payment Method
          </h4>
          <div />
          <div
            className="text"
          >
            PO
            : 
            123456
          </div>
        </div>
      </div>
    </div>
    <div
      className="cmp-order-details__order-summary"
      data-locator="order-summary"
    >
      <h4 />
      <div
        className="cmp-order-details__order-subtotal"
      >
        <div
          className="cmp-order-details__order-subtotal_left"
          data-locator="order-summary-label-sub-total"
        >
          Subtotal
           
          
        </div>
        <div
          className="cmp-order-details__order-subtotal_right"
          data-locator="order-summary-price-sub-total"
        >
          0000000000043.35
        </div>
      </div>
      <div
        className="cmp-order-details__order-savings"
      >
        <div
          className="cmp-order-details__order-savings_left"
          data-locator="order-summary-label-total-discount"
        >
          Savings
        </div>
        <div
          className="cmp-order-details__order-savings_right"
          data-locator="order-summary-price-total-discount"
        >
          -
          0000000000007.65
        </div>
      </div>
      <div
        className="cmp-order-details__order-shipping"
      >
        <div
          className="cmp-order-details__order-shipping_left"
          data-locator="order-summary-label-total-shipping-handling"
        >
          Shipping and Handling 
        </div>
        <div
          className="cmp-order-details__order-shipping_right"
          data-locator="order-summary-price-total-shipping-handling"
        >
          0000000000000
        </div>
      </div>
      <div
        className="cmp-order-details__order-tax"
      >
        <div
          className="cmp-order-details__order-tax_left"
          data-locator="order-summary-label-estimated-tax"
        >
          Tax
        </div>
        <div
          className="cmp-order-details__order-tax_right"
          data-locator="order-summary-price-estimated-tax"
        >
          0000000000000
        </div>
      </div>
      <div
        className="cmp-order-details__order-total"
      >
        <div
          className="cmp-order-details__order-total_left"
          data-locator="order-summary-label-total-price"
        />
        <div
          className="cmp-order-details__order-total_right"
          data-locator="order-summary-price-total-price"
        >
          <h1>
            0000000000043.35
          </h1>
        </div>
      </div>
      <div
        className="cmp-order-details__reorder"
        data-locator="order-details-reorder"
      >
        <a
          className="cmp-button"
          onClick={[Function]}
        >
          Reorder
        </a>
      </div>
    </div>
  </div>,
  <div
    className="cmp-order-details__order-shipment-list"
    data-locator="order-shipment-list"
  />,
  <div
    className="order-shipment__reorder"
    data-locator="order-shipment-reorder"
  >
    <a
      className="cmp-button"
      onClick={[Function]}
    >
      Reorder
    </a>
  </div>,
]
`;
